### PR TITLE
Fix logic of creating MTree snapshot

### PR DESCRIPTION
**Pull request by samperson1997**
Fix logic of creating MTree snapshot.
The snapshot should only be created when `System.currentTimeMillis() - logFile.lastModified() >= mtreeSnapshotThresholdTime` AND `logWriter.getLineNumber() >= mtreeSnapshotInterval`

### CODE REVIEW

File: server/src/main/java/org/apache/iotdb/db/metadata/MManager.java [hunk 1]
```diff
@@ -1785,27 +1785,25 @@ private void checkMTreeModified() {
       // the logWriter is not initialized now, we skip the check once.
       return;
     }
-    if (System.currentTimeMillis() - logFile.lastModified() >= mtreeSnapshotThresholdTime
-        && logWriter.getLineNumber() > 0) {
-      logger.info("Start creating MTree snapshot, because {} ms elapse.",
+    if (System.currentTimeMillis() - logFile.lastModified() < mtreeSnapshotThresholdTime) {
+      logger.info("MTree snapshot need not be created. Time from last modification: {} ms.",
```
Comment (jixuan1989) [inline server/src/main/java/org/apache/iotdb/db/metadata/MManager.java:1789]: how about debug level

File: server/src/main/java/org/apache/iotdb/db/metadata/MManager.java [hunk 2]
```diff
@@ -1785,27 +1785,25 @@ private void checkMTreeModified() {
...
           System.currentTimeMillis() - logFile.lastModified());
-      createMTreeSnapshot();
-    } else if (logWriter.getLineNumber() >= mtreeSnapshotInterval) {
-      logger.info("Start creating MTree snapshot, because of {} new lines are added.",
+    } else if (logWriter.getLineNumber() < mtreeSnapshotInterval) {
+      logger.info("MTree snapshot need not be created. New mlog line number: {}.",
```
Comment (jixuan1989) [inline server/src/main/java/org/apache/iotdb/db/metadata/MManager.java:1792]: how about debug level

File: server/src/main/java/org/apache/iotdb/db/metadata/MManager.java (hunk 1 above)
Comment (samperson1997) [reply server/src/main/java/org/apache/iotdb/db/metadata/MManager.java:1789]: Sure

File: server/src/main/java/org/apache/iotdb/db/metadata/MManager.java (hunk 2 above)
Comment (samperson1997) [reply server/src/main/java/org/apache/iotdb/db/metadata/MManager.java:1792]: Fixed

**Pull request merged**